modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/cxf/jbws4046/JBWS4046TestCase.java

```diff
@@ -0,0 +1,117 @@
+public class JBWS4046TestCase extends JBossWSTest {
+
+    @ArquillianResource
+    private URL baseURL;
+
+    @Deployment(name = "jbws4046", testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, "jbws4046.war");
+        archive.setManifest(new StringAsset("Manifest-Version: 1.0\n"
+                + "Dependencies: org.apache.cxf\n"))
+                .addClass(org.jboss.test.ws.jaxws.cxf.jbws4046.v1.Demo.class)
+                .addClass(org.jboss.test.ws.jaxws.cxf.jbws4046.v2.Demo.class)
+                .addClass(DemoServiceHandler.class)
+                .add(new FileAsset(new File(JBossWSTestHelper.getTestResourcesDir() + "/jaxws/cxf/jbws4046/handlers.xml")), "WEB-INF/classes/handlers.xml")
+                .setWebXML(new File(JBossWSTestHelper.getTestResourcesDir() + "/jaxws/cxf/jbws4046/WEB-INF/web.xml"));
+        return archive;
+    }
+
+    @Test
+    @RunAsClient
+    public void testServiceHandlerInitialization() throws Exception {
+        String v1Resp = callVersionService("1");
+        String v2Resp = callVersionService("2");
+
+        Assert.assertNotNull("Service version 1 failed", v1Resp);
+        Assert.assertNotNull("Service version 2 failed", v2Resp);
+    }
+
+    public String callVersionService(String path) throws SOAPException {
+        SOAPConnection soapConnection = null;
+        try {
+            soapConnection = SOAPConnectionFactory.newInstance().createConnection();
+
+            String serviceUrl = baseURL.toString() + "/" + path;
+            SOAPMessage soapResponse = soapConnection.call(serviceRequest(), serviceUrl);
+
+            return extractVersion(soapResponse);
+
+        } catch (Exception e) {
+            System.err.println("Error occurred while sending SOAP Request to Server");
+            e.printStackTrace();
+        } finally {
+            if (soapConnection != null)
+                soapConnection.close();
+        }
+        return null;
+    }
+
+    private SOAPMessage serviceRequest() throws Exception {
+        MessageFactory messageFactory = MessageFactory.newInstance();
+        SOAPMessage soapMessage = messageFactory.createMessage();
+        SOAPPart soapPart = soapMessage.getSOAPPart();
+        SOAPEnvelope envelope = soapPart.getEnvelope();
+
+        envelope.addNamespaceDeclaration("test", "http://org.jboss.ws/jaxws/cxf/jbws4046");
+        envelope.getBody().addChildElement("version", "test");
+
+        soapMessage.saveChanges();
+
+        return soapMessage;
+    }
+
+    private static String extractVersion(SOAPMessage soapResponse) throws Exception {
+        Iterator fooResponse = soapResponse.getSOAPBody().getChildElements(new QName("http://org.jboss.ws/jaxws/cxf/jbws4046", "versionResponse"));
+        while (fooResponse.hasNext()) {
+            return (((SOAPElement)fooResponse.next()).getChildNodes().item(0).getTextContent());
+        }
+        return null;
+    }
+}
```

Comment:
We can probably introduce an interface class to call this two webservice endpoints with jaxws api instead of saaj api. It could simplify the test code a bit . WDYT ?